### PR TITLE
Initial implementation and example of tests

### PR DESCRIPTION
**Pull request by apachano**
Authors:
@apachano 

## Summary of changes
- Add testing framework
- Add tests to inventory system
  
## Caveats
Does this introduce any new bugs?
- No

## On approval
Wait for me (need to figure out GH actions)

### CODE REVIEW

File: Phoenix/Common/CMakeLists.txt [hunk 1]
```diff
@@ -1,28 +1,57 @@
 project(PhoenixCommon)
 
+option(BUILD_TESTS OFF)
```
Comment (beeperdeeper089) [inline Phoenix/Common/CMakeLists.txt:3]: Do we want to rename this to PHX_BUILD_TESTS?

File: Phoenix/Common/CMakeLists.txt [hunk 2]
```diff
@@ -1,28 +1,57 @@
...
-	CXX_STANDARD_REQUIRED ON
-	CXX_EXTENSIONS OFF
-)
+		CXX_STANDARD 17
+		CXX_STANDARD_REQUIRED ON
+		CXX_EXTENSIONS OFF
+		)
+
+add_subdirectory(Test)
+add_executable(${PROJECT_NAME}_test ${Headers} ${Sources} ${Tests})
```
Comment (beeperdeeper089) [inline Phoenix/Common/CMakeLists.txt:30]: Can we move this stuff into another CMake file in the Tests directory?

File: Phoenix/Common/Test/Math/Vector3.test.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include <catch2/catch.hpp>
+
+#include <Common/Math/Vector3.hpp>
+
+using namespace phx::math::detail;
+
+TEST_CASE("Vec3 Addition", "[Vec3]")
+{
+	REQUIRE(Vector3<int> {1, 1, 1} + Vector3<int> {1, 1, 1} ==
```
Comment (beeperdeeper089) [inline Phoenix/Common/Test/Math/Vector3.test.cpp:9]: We need more test cases - probably randomised as well.

We need basic multiplication, division, addition and subtraction. I could probably fill that in for you at some point if you can't be bothered for this PR.

File: Phoenix/ThirdParty/CMakeLists.txt [hunk 4]
```diff
@@ -48,20 +49,20 @@ set(PHX_THIRD_PARTY_INCLUDES
...
+		Glad
+		ImGui
+		enet
+		sol2
+		liblua
+		nlohmann_json::nlohmann_json
+		$<$<PLATFORM_ID:Windows>:opengl32.lib> # link to opengl32.lib if windows.
+		$<$<PLATFORM_ID:Windows>:ws2_32.lib> # link to ws2_32.lib if windows.
+		$<$<PLATFORM_ID:Windows>:winmm.lib> # link to winmm.lib if windows.
+		#Catch2::Catch2
```
Comment (beeperdeeper089) [inline Phoenix/ThirdParty/CMakeLists.txt:63]: If it's just commented probably just remove it.

File: Phoenix/Common/CMakeLists.txt [hunk 5]
```diff
@@ -1,28 +1,57 @@
...
+option(BUILD_TESTS OFF)
+
 add_subdirectory(Include/Common)
 add_subdirectory(Source)
 
 add_library(${PROJECT_NAME} STATIC ${Headers} ${Sources})
 
 target_link_libraries(${PROJECT_NAME}
-	PRIVATE
+		PRIVATE
```
Comment (beeperdeeper089) [inline Phoenix/Common/CMakeLists.txt:11]: Your editor has replaced tab with 8 spaces so we probably want to fix that.

File: Phoenix/Common/CMakeLists.txt [hunk 6]
```diff
@@ -1,28 +1,57 @@
...
+		Include
+		)
+
+set_target_properties(${PROJECT_NAME}_test PROPERTIES
+		CXX_STANDARD 17
+		CXX_STANDARD_REQUIRED ON
+		CXX_EXTENSIONS OFF
+		)
+
+#if(BUILD_TESTS)
```
Comment (beeperdeeper089) [inline Phoenix/Common/CMakeLists.txt:52]: Remove if not needed.

However I'm assuming it's meant to disable/enable tests so it'd be nice if we can make that functional.

**Review by beeperdeeper089 — CHANGES_REQUESTED** [review]

File: Phoenix/Common/CMakeLists.txt (hunk 2 above)
Comment (apachano) [reply Phoenix/Common/CMakeLists.txt:30]: Not sure how with CMake - I can dig but initially trying to move it IDK how to get the headers and sources variable down to that file.

File: Phoenix/Common/CMakeLists.txt (hunk 1 above)
Comment (apachano) [reply Phoenix/Common/CMakeLists.txt:3]: Actually, I never ended up using this.

Comment (apachano) [reply Phoenix/Common/CMakeLists.txt:3]: I'mma implement it b/c that would make sense lol

File: Phoenix/Common/Test/Math/Vector3.test.cpp (hunk 3 above)
Comment (apachano) [reply Phoenix/Common/Test/Math/Vector3.test.cpp:9]: Agreed - but this was just a demo of "here is how you do shit"/ "this works". We will need to add some tickets to implement tests for existing code.

**Comment by apachano** [comment]
@beeperdeeper089 Fixed everything but moving the test code to the sub directory, it was throwing a fit when I tried.

File: Phoenix/Common/CMakeLists.txt (hunk 2 above)
Comment (beeperdeeper089) [reply Phoenix/Common/CMakeLists.txt]: It propogates, you can just do ${Headers} will work

File: Phoenix/Common/Test/Math/Vector3.test.cpp (hunk 3 above)
Comment (beeperdeeper089) [reply Phoenix/Common/Test/Math/Vector3.test.cpp:9]: Sounds good!

**Comment by beeperdeeper089** [comment]
I'll merge this for now, we can make a card for moving it in later. I might try get that done today if I can be bothered.

**Review by beeperdeeper089 — APPROVED** [review]

**Pull request merged**